### PR TITLE
oshmem: Fix preconnect and coverity

### PR DESCRIPTION
**Pull request by igor-ivanov**
Port 
https://github.com/open-mpi/ompi/pull/1169
https://github.com/open-mpi/ompi/pull/1178

:bot:label:bug
:bot:milestone:v2.0.0
:bot:assign: @hppritcha

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1094/ for details.

**Comment by mike-dubman** [comment]
:+1:

File: oshmem/info/info.c [hunk 1]
```diff
@@ -243,6 +255,10 @@ static int oshmem_info_get_heap_size(char *value, size_t *interp)
     if (size <= 0) {
         return OSHMEM_ERR_BAD_PARAM;
     } else {
+        opal_setenv(OSHMEM_ENV_SYMMETRIC_SIZE, p, true, &environ);
+        opal_setenv(SHMEM_HEAP_SIZE, p, true, &environ);
+/* Probably needless code */
+#if 0
```
Comment (hppritcha) [inline oshmem/info/info.c:261]: would be nice to really remove this if'd 0 code if not needed.

**Comment by hppritcha** [comment]
going ahead and merging.

**Pull request merged**